models/mesh.go

```diff
@@ -40,77 +40,75 @@ type Mesh struct {
 // your controlplane.
 type Tag struct {
 	// Cluster is the cluster that the tag is associated with.
-	Cluster string
-	// ControlPlane associated with the tag.
-	ControlPlane *ControlPlane
+	Cluster string `json:"cluster"`
 	// Name is the name of the tag.
-	Name string
+	Name string `json:"name"`
 	// Revision is the revision of the controlplane associated with this tag.
-	Revision string
+	Revision string `json:"revision"`
 }
 
 // ControlPlane manages the dataPlane for one or more kube clusters.
 // It's expected to manage the cluster that it is deployed in.
 // It has configuration for all the clusters/namespaces associated with it.
 type ControlPlane struct {
 	// Cluster the kube cluster that the controlplane is running on.
-	Cluster *KubeCluster
+	Cluster *KubeCluster `json:"cluster"`
 
 	// Config
-	Config ControlPlaneConfiguration
+	Config ControlPlaneConfiguration `json:"config"`
 
 	// ExternalControlPlane indicates if the controlplane is managing an external cluster.
-	ExternalControlPlane bool
+	ExternalControlPlane bool `json:"externalControlPlane"`
 
 	// ID is the control plane ID as known by istiod.
-	ID string
+	ID string `json:"id"`
 
 	// IstiodName is the control plane name
-	IstiodName string
+	IstiodName string `json:"istiodName"`
 
 	// IstiodNamespace is the namespace name of the deployed control plane
-	IstiodNamespace string
+	IstiodNamespace string `json:"istiodNamespace"`
 
 	// ManagedClusters are the clusters that this controlplane manages.
 	// This could include the cluster that the controlplane is running on.
-	ManagedClusters []*KubeCluster
+	ManagedClusters []*KubeCluster `json:"managedClusters"`
 
 	// ManagesExternal indicates if the controlplane manages an external cluster.
 	// It could also manage the cluster that it is running on.
-	ManagesExternal bool
+	ManagesExternal bool `json:"managesExternal"`
 
 	// ManagedNamespaces are the namespaces that the controlplane is managing.
 	// More specifically, it is a namespace with either injection enabled
 	// or ambient enabled and it matches this controlplane's revision either
 	// directly or through a tag.
-	ManagedNamespaces []Namespace
+	ManagedNamespaces []Namespace `json:"managedNamespaces"`
 
 	// Resources are the resources that the controlplane is using.
-	Resources corev1.ResourceRequirements
+	Resources corev1.ResourceRequirements `json:"resources"`
 
 	// Revision is the revision of the controlplane.
 	// Can be empty when it's the default revision.
-	Revision string
+	Revision string `json:"revision"`
 
 	// Status is the status of the controlplane as reported by kiali.
 	// It includes the deployment status and whether kiali can connect
 	// to the controlplane or not.
-	Status string
+	Status string `json:"status"`
 
 	// Tags are the tags associated with the controlplane.
-	Tags []Tag
+	Tag *Tag `json:"tag"`
```

Comment:
```suggestion
	Tag *Tag `json:"tag,omitempty"`
```

I think this and several others could probably have have omitempty, and save us from nulls in the json.